Code/Framework/AzCore/AzCore/Module/Environment.cpp

```diff
@@ -75,6 +76,45 @@ namespace AZ
         bool operator==(const OSStdAllocator& a, const OSStdAllocator& b) { (void)a; (void)b; return true; }
         bool operator!=(const OSStdAllocator& a, const OSStdAllocator& b) { (void)a; (void)b; return false; }
 
+        void EnvironmentVariableHolderBase::UnregisterAndDestroy(DestructFunc destruct, bool moduleRelease)
+        {
+            bool releaseByModule = false;
+            const bool releaseByUseCount = (--m_useCount == 0);
+            // We take over the lock, and release it before potentially destructin/freeing ourselves
```

Comment:
minor: "destructin" should be "destroying"

Reply:
Indeed, it should and now is :smile: